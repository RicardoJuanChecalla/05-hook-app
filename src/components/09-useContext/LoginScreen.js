import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const {setUser} = useContext(UserContext);
  const userNew = {
    id:123,
    name:'Ricardo'
  }
  return <div>
    <h1>LoginScreen</h1>
    <hr/>
    <button 
      className='btn btn-primary'
      onClick={()=>setUser(userNew)}
    >Login</button>
  </div>;
};
