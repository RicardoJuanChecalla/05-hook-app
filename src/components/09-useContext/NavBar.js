import React from 'react';
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
    // <nav>
    //   <ul>
    //       <li>
    //           <Link to='/'>Home</Link>
    //       </li>
    //       <li>
    //           <Link to='/about'>About</Link>
    //       </li>
    //       <li>
    //           <Link to='/login'>Login</Link>
    //       </li>
    //   </ul>
    // </nav>
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to='/'>useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav">
                <NavLink to='/' className="nav-item nav-link">Home</NavLink>
                <NavLink to='/about' className="nav-item nav-link">About</NavLink>
                <NavLink to='/login' className="nav-item nav-link">Login</NavLink>
            </div>
            </div>
    </nav>
  );
};
