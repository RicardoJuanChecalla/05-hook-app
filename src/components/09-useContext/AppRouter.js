import React from 'react';
import { BrowserRouter as Router, Route, Routes, Navigate } from 'react-router-dom'
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    
  return (
    <Router>
        <div>
            <NavBar/>
            <div className='container'>
              <Routes>
                  {/* <Route path='/about' component={AboutScreen}/> */}
                  {/* <Route path="/about" component={AboutScreen} /> */}
                  <Route exact path="/about" element={<AboutScreen />} />
                  <Route exact path="/login" element={<LoginScreen />} />
                  <Route exact path="/" element={<HomeScreen />} />
                  <Route path="*" element={<Navigate to="/" />}/>
              </Routes>
            </div>
        </div>
    </Router>
  );
};
