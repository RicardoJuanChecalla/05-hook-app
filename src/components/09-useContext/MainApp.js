import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    // const user = {
    //     id: 123,
    //     name: 'Ricardo',
    //     email:'rchecalla@cmactacna.com.pe'
    // }
    const [user, setUser] = useState({});
    return (
        <UserContext.Provider value={{user, setUser}}>
            <AppRouter/>
        </UserContext.Provider>
    );
//   return <div>
//     <h1>MainApp</h1>
//     <hr/> 
//     <AppRouter/>
//   </div>;
};
