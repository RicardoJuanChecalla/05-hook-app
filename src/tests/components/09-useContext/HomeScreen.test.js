import { mount } from 'enzyme';
import '@testing-library/jest-dom';
import { demoTodos } from '../../fixtures/demoTodos';
import { act } from '@testing-library/react';
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en  <HomeScreen />', () => {
    const user = {
        name: 'Ricardo',
        email: 'rchecalla@cmactacna.com.pe'
    }
    const wrapper = mount(
        <UserContext.Provider value={{user:user}}>
            <HomeScreen />
        </UserContext.Provider>  
    );
    
  test('Debe mostrarse correctamente', () => {
   expect(wrapper).toMatchSnapshot();
  });
});
