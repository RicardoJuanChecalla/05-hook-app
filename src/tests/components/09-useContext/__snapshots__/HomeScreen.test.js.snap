// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <HomeScreen /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Ricardo",
   "email": "rchecalla@cmactacna.com.pe"
}
  </pre>
</div>
`;
